### PR TITLE
Refactor: Move functions unrelated to printing from printer-estree.js to utils.js

### LINKED ISSUE #6513: Refactor proposal: Move functions unrelated to print to `utils.js` from `printer-estree.js`.
(Sorry, this is not bug report, so I didn't follow to an issue template)

I noticed to there are functions unrelated to printing, when I was reading `language-js/printer-estree.js`. I think that these functions should be defined in `language-js/utils.js`. For example, `isBinaryish`, `isMemberish`, `isNumricLiteral`, `isStringLiteral`, and more..

How do you think about it?

If it is needed, I'll do it.

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fix #6513 

This PR is just moving functions, does not change content of functions.

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
@evilebottnawi Should I merge PR by myself? Or should I wait for other maintainers merge?

**Comment by evilebottnawi** [comment]
@sosukesuzuki it would be great to have at least 3 approves before merging, i think we should setup this in repo settings

**Comment by duailibe** [comment]
I don't like this, these functions are only used in the printer file, they are only relevant to the printing context, but shrug

**Comment by lydell** [comment]
I’m also a bit _meh_ on this. Feels like we maybe make `git blame` harder to use here for not much gain? ¯\\\_(ツ)\_/¯

**Comment by duailibe** [comment]
Just read the issue you opened

> If it is needed, I'll do it.

I don't think it is needed

**Comment by evilebottnawi** [comment]
@duailibe we use same approach for other printer like css/markdown/etc i think using same structure will be great

**Comment by evilebottnawi** [comment]
Also it is really hard to read big file, also mine IDE very slow on this file (but it is my problem, i think)

**Comment by lydell** [comment]
It's not the first time I hear "my IDE is slow no this file" as an argument for splitting a file, though.

**Comment by sosukesuzuki** [comment]
Personally, I thought that `printer-estree.js` was too big for Prettier beginner to understand, and the functions in `printer-estree.js` were general enough, so I created this PR.

(However, I don't have a strong will to make this change. I can close this PR if it is unnecessary for other maintainers.)

File: src/language-js/utils.js [hunk 1]
```diff
@@ -107,11 +118,832 @@ function getLeftSidePathName(path, node) {
...
+  isFlowAnnotationComment,
+  hasLeadingOwnLineComment,
+  returnArgumentHasLeadingComment,
+  isStringPropSafeToCoerceToIdentifier,
+  isJestEachTemplateLiteral,
+  isTemplateOnItsOwnLine,
+  needsHardlineAfterDanglingComment,
+  conditionalExpressionChainContainsJSX,
+  rawText,
+  identity
```
Comment (lipis) [inline src/language-js/utils.js]: sort by name

Comment (sosukesuzuki) [reply src/language-js/utils.js]: Defined order? Alphabetical order?

Comment (lipis) [reply src/language-js/utils.js]: yes.. whatever VS Code is doing for sorting lines :)

File: src/language-js/utils.js [hunk 2]
```diff
@@ -107,11 +118,832 @@ function getLeftSidePathName(path, node) {
...
+}
+
+function hasDanglingComments(node) {
+  return (
+    node.comments &&
+    node.comments.some(comment => !comment.leading && !comment.trailing)
+  );
+}
+
+/** identify if an angular expression seems to have side effects */
```
Comment (lipis) [inline src/language-js/utils.js]:
```suggestion
// Identify if an angular expression seems to have side effects 
```

**Comment by lipis** [comment]
Let's close this one until further notice :)

**Comment by evilebottnawi** [comment]
@lipis why? I think it is good impovement

**Comment by lipis** [comment]
I don't have a strong opinion, and most of the other maintainers said it's not necessary.

**Comment by evilebottnawi** [comment]
@lipis only one maintainer against it, just note - other printers use structure with `utils.js`

**Comment by evilebottnawi** [comment]
It is really hard to read and slow in many IDE, file are very big

**Comment by lipis** [comment]
Then let's get one more approval and merge it.

**Comment by lipis** [comment]
@sosukesuzuki can you fix the conflicts?

**Comment by sosukesuzuki** [comment]
I'll merge this pr

**Comment by evilebottnawi** [comment]
need wait @duailibe

**Pull request merged**